crates/matrix-sdk-crypto/src/olm/group_sessions/inbound.rs

```diff
@@ -55,6 +55,26 @@ use crate::{
     },
 };
 
+/// The trust level required to decrypt an event
+#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+pub enum TrustRequirement {
+    /// Decrypt events from everyone regardless of trust
+    Untrusted,
+    /// Only decrypt events from cross-signed or legacy devices
+    CrossSignedOrLegacy,
+    /// Only decrypt events from cross-signed devices
+    CrossSigned,
+    /// Only decrypt events from cross-signed devices where we have verified the user
+    VerifiedUser,
```

Comment:
Not sure, but perhaps it makes sense to call it user identity instead of user, since it might confuse people that those are the same concept.
```suggestion
    VerifiedUserIdentity,
```